### ISSUE TITLE
Bumping into a platform from bellow may allow jumping again

### ISSUE THREAD

**Issue by arthuro555** [4ian/GDevelop#2386]
## Describe the bug
With some platforms, when bumping from bellow, the platformer character gets put in the state "not jumping".
For example, if I have a game with those events:
![image](https://user-images.githubusercontent.com/19349038/109120612-76d4b680-7746-11eb-97af-9fc871ce63c8.png)
Normal Behavior is you jump, and you stay jumping until you touch the ground:
![JoFEpniNgZ](https://user-images.githubusercontent.com/19349038/109121143-2b6ed800-7747-11eb-9ec9-ca5260b1422a.gif)
But on this platform bumping into the platform changes the state and allows jumping again:
![dKWxg2ESTN](https://user-images.githubusercontent.com/19349038/109121402-7b4d9f00-7747-11eb-9f72-bdaa9babf707.gif)

## To Reproduce
1. Download and open [this project file](https://github.com/4ian/GDevelop/files/6041378/PlatformerExample.zip) in GDevelop.
2. Launch a preview.
3. Go to the right part of the stage.
4. Bump into the ceiling like in the gif above and see the bug.

## Other details
 - Windows 10
 - Beta 105
 - electron-app

**Comment by 4ian** [comment]
You forgot to mention that you have some logic to allow to jump again, by doing a double jump ;)

The problem is because the state "jumping" is only for the first part of the jump. When you reach the top, you start to be in the "fall" state.
In the logic you have, you check if the player is jumping, and if yes, you reset the variable to allow a double jump after 0.25 seconds.

But if you collide with the ceiling, you'll go from jump (do double jump is activated) to fall, and as double jump is still allowed, you trigger the double jump, so we're in jumping state again... and because the trigger once was broken (during the fall part), you can jump again.

To fix this: You can allow the double jump like you did a few seconds after the player is in the jump state (i.e: canDJump = 1), but when the double jump is done, don't set it back to 0. Set it to 2 ("double jump done") and only set it back to 0 when the player is on the floor.

Basically:
0 -> on the floor, double jump not "allowed".
1 -> in the air after a jump, can double jump after the timer is elapsed.
2 -> made a double jump. The next one will be allowed when reaching the floor.

**Comment by arthuro555** [comment]
Ah yes I am sorry. I thought this was a bug because in the first gif the jumping state ended so late that I thought the jumping condition would still trigger in the falling state.
